ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -302,11 +302,57 @@ func convertToECSVolumes(hostPaths *adapter.Volumes) []*ecs.Volume {
 			}}
 		output = append(output, ecsVolume)
 	}
-	// volumes with an empty host path
-	for _, volName := range hostPaths.VolumeEmptyHost {
+
+	// process docker volumes
+	dockerVolumes := make(map[string]DockerVolume)
+	if ecsParams != nil {
+		for _, dockerVol := range ecsParams.TaskDefinition.DockerVolumes {
+			if dockerVol.Name != "" {
+				dockerVolumes[dockerVol.Name] = dockerVol
+			}
+		}
+	}
+
+	// volumes without host path (allowed to have Docker Volume Configuration)
+	output = append(output, mergeVolumesWithoutHost(hostPaths.VolumeEmptyHost, ecsParams)...)
+	return output
+}
+
+func mergeVolumesWithoutHost(composeVolumes []string, ecsParams *ECSParams) []*ecs.Volume {
+	volumesWithoutHost := make(map[string]DockerVolume)
+	output := []*ecs.Volume{}
+
+	for _, volName := range composeVolumes {
+		volumesWithoutHost[volName] = DockerVolume{}
+	}
+
+	if ecsParams != nil {
+		for _, dockerVol := range ecsParams.TaskDefinition.DockerVolumes {
+			if dockerVol.Name != "" {
+				log.Info("is it right here:")
+				log.Info(dockerVol)
+				volumesWithoutHost[dockerVol.Name] = dockerVol
+			}
+		}
+	}
+
+	for volName, dVol := range volumesWithoutHost {
 		ecsVolume := &ecs.Volume{
 			Name: aws.String(volName),
 		}
+		if dVol.Name != "" {
```

Comment:
Do we have validation before this that any volume specified in `ecs-params` includes a `Name` (or other required fields) ?

Reply:
...and can we add a test for error case(s) handling if/when we do add validation?

Reply:
Good point